host-macros/src/service.rs

```diff
@@ -198,23 +256,34 @@ impl ServiceBuilder {
                 vis: ch.vis.clone(),
                 mutability: syn::FieldMutability::None,
             });
+            doc_strings.push(ch.args.doc_string.to_owned());
 
             // At least two attributes will be added to the attribute table for each characteristic:
             // - The characteristic declaration
             // - The characteristic's value declaration
             //
-            // If the characteristic has either the notify or indicate property, a Client Characteristic Configuration Descriptor (CCCD) declaration will also be added.
+            // If the characteristic has either the notify or indicate property,
+            // a Client Characteristic Configuration Descriptor (CCCD) declaration will also be added.
             self.attribute_count += if ch.args.notify || ch.args.indicate { 3 } else { 2 };
 
             self.construct_characteristic_static(ch);
         }
-
         // Processing common to all fields
-        for field in fields {
+        for (field, doc_string) in fields.iter().zip(doc_strings) {
+            let docs: TokenStream2 = doc_string
+                .lines()
+                .map(|line| {
+                    let span = field.span();
+                    quote_spanned!(span=>
+                        #[doc = #line]
```

Comment:
Oh I see, you're removing the syntactic sugar - is that why the docstrings weren't showing up in the tooltips?

Reply:
I'm not sure, they still show up as sugared in the expanded output.

Reply:
Huh.